telemetry/statsd/src/main/scala/io/buoyant/telemetry/StatsDInitializer.scala

```diff
@@ -0,0 +1,56 @@
+package io.buoyant.telemetry
+
+import com.timgroup.statsd.NonBlockingStatsDClient
+import com.twitter.finagle.Stack
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.logging.Logger
+import io.buoyant.telemetry.statsd.{StatsDStatsReceiver, StatsDTelemeter}
+import java.util.UUID
+
+class StatsDInitializer extends TelemeterInitializer {
+  type Config = StatsDConfig
+  val configClass = classOf[StatsDConfig]
+  override val configId = "io.l5d.statsd"
+}
+
+private[telemetry] object StatsDConfig {
+  val DefaultPrefix = "linkerd"
+  val DefaultHostname = "127.0.0.1"
+  val DefaultPort = 8125
+  val DefaultGaugePeriodMs = 10000
+
+  val MaxQueueSize = 10000
+  val ProcessId = "l5d-uuid-" + UUID.randomUUID().toString
+}
+
+case class StatsDConfig(
+  prefix: Option[String],
+  hostname: Option[String],
+  port: Option[Int],
+  gaugePeriodMs: Option[Int]
+) extends TelemeterConfig {
+  import StatsDConfig._
+
+  private[this] val log = Logger.get("io.l5d.statsd")
+
+  // prefix format is: "prefix_l5d-uuid-process-id"
+  val statsDPrefix = prefix.getOrElse(DefaultPrefix) + "_" + ProcessId
```

Comment:
I would suggest leaving this to user, user can always generate machine specific prefix (using sed for example, per instance).

User probably will encode things like this in prefix, by doing `sed` for the config file.

l5d.${stage}.${service}.${instanceId}